R/summary_funs.R

```diff
@@ -202,198 +209,223 @@ weighted_summary_means <- function(y_wobs_test, family, wdraws, mu, dis, cl_ref,
-                            alpha = alpha, ...)
-        res_diff <- get_stat(summ$mu, summ$lppd, varsel$y_wobs_test, stat,
-                             mu.bs = summ_ref$mu, lppd.bs = summ_ref$lppd,
-                             wcv = summ$wcv, alpha = alpha, ...)
-        val <- res_ref$value + res_diff$value
-        # TODO (subsampled PSIS-LOO CV): Is `val.se` really computed correctly
-        # or do we need to take into account that `res_ref$se` and `res_diff$se`
-        # might be stochastically dependent?
-        val.se <- sqrt(res_ref$se^2 + res_diff$se^2)
-        if (stat %in% c("rmse", "auc")) {
-          # TODO (subsampled PSIS-LOO CV): Use bootstrap for lower and upper
-          # confidence interval bounds as well as for the standard error.
-          warning("Lower and upper confidence interval bounds of performance ",
-                  "statistic `", stat, "` are based on a normal ",
-                  "approximation, not the bootstrap. The standard error of ",
-                  "performance statistic `", stat, "` is also not based on a ",
-                  "bootstrap.")
-        }
-        lq <- qnorm(alpha / 2, mean = val, sd = val.se)
-        uq <- qnorm(1 - alpha / 2, mean = val, sd = val.se)
-        row <- data.frame(
-          data = varsel$type_test, size = k - 1, delta = delta,
-          statistic = stat, value = val, lq = lq, uq = uq, se = val.se,
-          diff = res_diff$value, diff.se = res_diff$se
-        )
+    for (k in seq_along(summaries_sub)) {
+      diff <- get_stat(summaries = summaries_sub[[k]],
+                       summaries_baseline = summaries_baseline,
+                       summaries_fast = summaries_fast_sub[[k]],
+                       varsel$y_wobs_test, stat, alpha = alpha, ...)
+      if (!delta) {
+        res <- get_stat(summaries = summaries_sub[[k]],
+                        summaries_baseline = NULL,
+                        summaries_fast = summaries_fast_sub[[k]],
+                        varsel$y_wobs_test, stat, alpha = alpha, ...)
       } else {
-        ## normal case
-        res <- get_stat(summ$mu, summ$lppd, varsel$y_wobs_test, stat,
-                        mu.bs = mu.bs, lppd.bs = lppd.bs, wcv = summ$wcv,
-                        alpha = alpha, ...)
-        diff <- get_stat(summ$mu, summ$lppd, varsel$y_wobs_test, stat,
-                         mu.bs = summ_ref$mu, lppd.bs = summ_ref$lppd,
-                         wcv = summ$wcv, alpha = alpha, ...)
-        row <- data.frame(
-          data = varsel$type_test, size = k - 1, delta = delta,
-          statistic = stat, value = res$value, lq = res$lq, uq = res$uq,
-          se = res$se, diff = diff$value, diff.se = diff$se
-        )
+        res <- diff
       }
+      row <- data.frame(
+        data = varsel$type_test, size = k - 1, delta = delta,
+        statistic = stat, value = res$value, lq = res$lq, uq = res$uq,
+        se = res$se, diff = diff$value, diff.se = diff$se
+      )
```

Comment:
I think there might be something wrong here. If (and only if) `delta` is `FALSE`, then `summaries_baseline` is `NULL` (see definition of `delta <- FALSE` further above). Thus, if `delta` is `FALSE`, object `diff` won't contain any difference estimates. Then, in the `if` part, if `delta` is `FALSE`, then object `res` will be identical to `diff` and hence the creation of `ref` in lines `res <- get_stat([...])` could be avoided, but more importantly, `ref` also won't contain any difference estimates. Also note that before these changes, elements `diff` and `diff.se` of `row` always took the reference model as the baseline model. This does not seem to hold anymore.

Reply:
Hmm... All stats work with and without subsampling, and with and without deltas=TRUE (except AUC is not implemented with subsampling and falls back to fast LOO). I don't know what would be the test case where there should be a problem with this

Reply:
Given that I did not have your change concerning `deltas = TRUE` in mind (still showing original-scale results for non-`**pd` statistics), your code might in fact be correct. I need to think about this some other day again.

Reply:
I thought about this again, but I still think my remarks from [above](https://github.com/stan-dev/projpred/pull/496#discussion_r1650065426) hold. Example:
```r
data("df_gaussian", package = "projpred")
dat <- data.frame(y = df_gaussian$y, df_gaussian$x)
rfit <- rstanarm::stan_glm(y ~ X1 + X2 + X3 + X4 + X5,
                           data = dat,
                           chains = 1,
                           iter = 500,
                           seed = 1140350788,
                           refresh = 0)

devtools::load_all(".")

vs <- varsel(rfit,
             nclusters = 3,
             nclusters_pred = 5,
             nterms_max = 2,
             seed = 46782345)

summary(vs)

```
gives (at the current state of this PR, i.e., commit c23fa783b1dc044cbe4e52e25682e530192f6fde):
```

Family: gaussian 
Link function: identity 

Formula: y ~ X1 + X2 + X3 + X4 + X5
Observations: 100
Projection method: traditional
Search method: forward
Maximum submodel size for the search: 2
Number of projected draws in the search: 3 (from clustered projection)
Number of projected draws in the performance evaluation: 5 (from clustered projection)
Argument `refit_prj`: TRUE

Submodel performance evaluation summary with `deltas = FALSE` and `cumulate = FALSE`:
 size ranking_fulldata cv_proportions_diag elpd elpd.se elpd.diff elpd.diff.se
    0      (Intercept)                  NA -249     5.3      -249          5.3
    1               X1                  NA -231     5.6      -231          5.6
    2               X5                  NA -220     6.0      -220          6.0

Reference model performance evaluation summary with `deltas = FALSE`:
   elpd elpd.se 
 -210.1     6.6
```
You can see that for the submodels, `elpd.diff` is the same as `elpd` and not the ELPD-diff vs. the reference model.

Reply:
Marking this conversation as "resolved" because this problem has been circumvented (for this PR) by moving the new "mixed deltas" plot, the new column behavior of `summary.vsel()`, and some other changes to branch [`mixed_deltas_plot`](https://github.com/stan-dev/projpred/tree/mixed_deltas_plot).